### ISSUE TITLE
GLFW_MAXIMIZED  hint not working on Windows in version 3.3

### ISSUE THREAD

**Issue by dougbinks** [glfw/glfw#1499]
Reported on our forums in post [Maximising Window differences from 3.2 to 3.3 ](http://discourse.glfw.org/t/maximising-window-differences-from-3-2-to-3-3/1338).

I have confirmed the issue and have a fix in testing which I'll create a PR for shortly, however this turns out to be a slightly complex and subtle area.

**Details**

When the window hint GLFW_MAXIMIZED is set to GLFW_TRUE and a window is created it appears with maximized style but not with a maximized size.

It should instead appear maximized as well as in the maximized style.

**Notes**

Related to issue #780.

**Comment by dougbinks** [comment]
The simplest solution I can see is to alter _glfwPlatformShowWindow() to:

```
void _glfwPlatformShowWindow(_GLFWwindow* window)
{
	if (window->win32.maximized)
		ShowWindow(window->win32.handle, SW_MAXIMIZE);
	else
		ShowWindow(window->win32.handle, SW_SHOWNA);
}
```

Currently this means any maximized window would then be focused, as SW_MAXIMIZE will activate the window and there is no maximize without activate option that I can see. This makes sense as a newly created maximized window would cover all other windows.

**Comment by elmindreda** [comment]
This should be fixed now with 711b9694a133ef338482ae75ada724e3a9915d91. Thank you for reporting this!
